server/testsupport/actionPlanTestDataBuilder.ts

```diff
@@ -1,6 +1,34 @@
 import type { Step, Goal, ActionPlan } from 'viewModels'
 import moment from 'moment'
 
+const aValidActionPlan = (options?: {
+  prisonNumber?: string
+  goals?: Array<Goal>
+  problemRetrievingData?: boolean
+}): ActionPlan => {
+  return {
+    prisonNumber: options?.prisonNumber || 'A1234BC',
+    goals: options?.goals || [aValidGoal()],
+    problemRetrievingData:
+      !options || options.problemRetrievingData === null || options.problemRetrievingData === undefined
+        ? true
+        : options.problemRetrievingData,
+  }
+}
+
+/**
+ * @deprecated - Use `aValidActionPlan` above because specifying a different collection of goals with this method makes
+ * for code that is confusing to read and reason about.
+ * EG:
+ * ```
+ * // create an action plan with 3 goals
+ * const actionPlan = aValidActionPlanWithOneGoal(
+ *  {
+ *    goals: [aValidGoal(), aValidGoal(), aValidGoal()],
+ *  }
+ * )
+ * ```
+ */
 const aValidActionPlanWithOneGoal = (
```

Comment:
In this PR I decided to deprecate this method with a suitable comment rather than rename it, otherwise this PR would have a load of unrelated change where I'd renamed this method.
We can remove this deprecated method and change all the existing uses of it to `aValidActionPlan()` in a separate PR

Reply:
I may have missed something, but `aValidActionPlanWithOneGoal()` seems ok to me?! Perhaps the goals array argument could be changed to a single `goal`, so it can be changed by a test if necessary, but otherwise I think it's ok! :)

Reply:
I think that was the problem though - the method name says its an action plan with one goal, but the method arg allows you to specify any number of goals. And I was after setting up an action plan with no goals, so I really didnt want to use this method.

But you are right that instead of deprecating the method we could change the method arg to only take a single goal. Leave it with me  ..... 👍